src/routing_message_filter.rs

```diff
@@ -70,9 +72,15 @@ impl RoutingMessageFilter {
 
     // Filter outgoing `RoutingMessage`. Return whether this specific message has been seen recently
     // (and thus should not be sent, due to deduplication).
+    //
+    // Return `false` if serialisation of the message fails - that can be handled elsewhere.
```

Comment:
I will prefer the approach of `return a Result<bool, ...>`.
Also I agree this can be done in a later PR.